File: LeetCode/189.RotateArrays.py
from typing import List


def RotateArray(nums: List[int], k: int):
    temp = [0] * len(nums)
    for i in range(len(nums)):
        idx = (i + k) % len(nums)
        temp[idx] = nums[i]
    nums.clear()
    nums.extend(temp)

arr = [1,2,3,4,5,6,7]
RotateArray(arr, 3)
print(arr)

arr = [-1, -100, 3, 99]
RotateArray(arr, 2)
print(arr)

File: ArrayProblems/189.RotateArrays.js
function RotateArray(nums, k) {
    let temp = new Array(nums.length)
    for (let i = 0; i < nums.length; i++) {
        const idx = ( i + k ) % nums.length;
        temp[idx] = nums[i];
    }
    for (let i = 0; i < nums.length; i++) {
        nums[i] = temp[i]
    }
}

let arr = [1,2,3,4,5,6,7]
RotateArray(arr, 2)
console.log(arr);